src/document.rs

```diff
@@ -47,6 +47,38 @@ fn yaml_to_liquid(yaml: &Yaml) -> Option<Value> {
     }
 }
 
+/// Escape HTML string
+///
+/// The function has been extracted from `librustdoc/html/escape.rs`.
+/// See also http://stackoverflow.com/questions/7381974 for details.
+pub fn escape_html(html: &str) -> String {
+    let mut escaped = String::new();
+    let mut last = 0;
+    for (i, ch) in html.bytes().enumerate() {
+        match ch as char {
+            '<' | '>' | '&' | '\'' | '"' => {
+                escaped.push_str(&html[last..i]);
+                let s = match ch as char {
+                    '>' => "&gt;",
+                    '<' => "&lt;",
+                    '&' => "&amp;",
+                    '\'' => "&#39;",
+                    '"' => "&quot;",
+                    _ => unreachable!(),
+                };
+                escaped.push_str(s);
+                last = i + 1;
+            }
+            _ => {}
+        }
+    }
+
+    if last < html.len() {
+        escaped.push_str(&html[last..]);
+    }
+    escaped
+}
```

Comment:
We should consider publishing a crate with this functionality. We will also need it in liquid to add an `escape` filter and I've encountered multiple occasions where a project needed a simple HTML escape function. An alternative could be using e.g. https://github.com/nagisa/marksman_escape/